tests/e2e-evm/test/connect.test.ts

```diff
@@ -0,0 +1,50 @@
+import hre from "hardhat";
+import { expect } from "chai";
+import { whaleAddress, userAddress } from "./addresses";
+
+describe("Viem Setup", function () {
+  it("getChainID() matches the configured network", async function () {
+    const publicClient = await hre.viem.getPublicClient();
+
+    const expectedChainId = (() => {
+      switch (hre.network.name) {
+        case "hardhat":
+          return 31337;
```

Comment:
Should we extract all those version and other constants (like chain versions, private keys, addresses, etc into one file so it contains them once with all necessary comments?